core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerImpl.java

```diff
@@ -187,9 +203,17 @@ public void pause() {
     }
 
     @Override
-    public void loadVideo(Uri uri, ContentType contentType) {
+    public void loadVideo(final Uri uri, ContentType contentType) {
+        if (mediaPlayer.hasPlayedContent()) {
+            reset();
+        }
         listenersHolder.getBufferStateListeners().onBufferStarted();
-        mediaPlayer.prepareVideo(uri);
+        requestSurface(new SurfaceHolderRequester.Callback() {
```

Comment:
I wonder if this is a better place to `requestAudioFocus`. Given the symmetry with `requestSurface`. 

Probably out of the scope of this PR